dart/utils/PackageResourceRetriever.cpp

```diff
@@ -71,32 +71,37 @@ void PackageResourceRetriever::addPackageDirectory(
 }
 
 //==============================================================================
-bool PackageResourceRetriever::exists(const std::string& _uri)
+bool PackageResourceRetriever::exists(const common::Uri& _uri) const
 {
   std::string packageName, relativePath;
   if (!resolvePackageUri(_uri, packageName, relativePath))
     return false;
 
-  for(const std::string &packagePath : getPackagePaths(packageName))
+  for(const std::string& packagePath : getPackagePaths(packageName))
   {
-    const std::string localUri = "file://" + packagePath + relativePath;
-    if (mLocalRetriever->exists(localUri))
+    common::Uri fileUri;
+    fileUri.fromPath(packagePath + relativePath);
```

Comment:
This should use the `fromRelativeUri` method to join `packagePath` and `relativePath`, not direct string concatenation.